### PR TITLE
Fix/logged redirect

### PR DESCRIPTION
**Pull request by rafarubim**
#### What is the purpose of this pull request?
Redirect user if he's logged in

#### What problem is this solving?
This also refactors the redirects in the app to improve the code

#### How should this be manually tested?

https://rafaprtest2--storecomponents.myvtex.com/. Sign in, then access https://rafaprtest2--storecomponents.myvtex.com/login and you should be redirected home.

You can also test manually by running `yarn test` in `react/` folder

#### Types of changes
- [X] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

[ch35436]

### CODE REVIEW

File: react/components/LoginContent.js [hunk 1]
```diff
@@ -248,16 +248,12 @@ class LoginContent extends Component {
...
-   */
+  
   onLoginSuccess = () => {
-    const { loginCallback } = this.props
+    const { isHeaderLogin } = this.props
     return this.context.patchSession().then(() => {
-      if (loginCallback) {
-        loginCallback()
+      if (isHeaderLogin) {
+        window && window.location && window.location.reload()
```
[the remark this answers is not in the document]
Comment (rnbez) [reply react/components/LoginContent.js:248]: Any chance this gets executed on server side?

File: react/components/LoginContent.js [hunk 2]
```diff
@@ -330,7 +325,10 @@ class LoginContent extends Component {
     // the profile by the props and current endpoint are /login, if receive it, should render the account options.
     if (sessionProfile && !profile) {
       if (location.pathname.includes('/login')) {
```
Comment (rnbez) [inline react/components/LoginContent.js:331]: The `runtime` has a prop with the page name. You should look into that later.

File: react/components/LoginContent.js [hunk 3]
```diff
@@ -449,9 +448,12 @@ const LoginContentProvider = props => {
...
         location: { pathname, search },
       },
+      query : {
+        returnUrl: returnUrlQueryString,
+      },
     } = runtime
-    const currentUrl = page !== 'store.login' ? `${pathname}${search}` : '/'
-    return path(['query', 'returnUrl'], props) || currentUrl
+    const currentUrl = page !== 'store.login' ? `${pathname}${search}` : '/' 
+    return returnUrlQueryString || currentUrl
```
Comment (rnbez) [inline react/components/LoginContent.js]: Any chance `query` can be null?

File: react/components/LoginContent.js [hunk 4]
```diff
@@ -452,8 +452,10 @@ const LoginContentProvider = props => {
         returnUrl: returnUrlQueryString,
       },
     } = runtime
-    const currentUrl = page !== 'store.login' ? `${pathname}${search}` : '/' 
-    return returnUrlQueryString || currentUrl
+    if (returnUrlQueryString) {
+      return returnUrlQueryString
+    }
+    return page !== 'store.login' ? `${pathname}${search}` : '/'
```
Comment (rnbez) [inline react/components/LoginContent.js]: rootpath?

File: react/components/LoginContent.js [hunk 5]
```diff
@@ -465,7 +462,11 @@ const LoginContentProvider = props => {
           </div>
         }
         return (
-        <LoginContentWrapper {...props} intl={intl} runtime={runtime} returnUrl={redirectUrl} />
+          <AuthServiceLazy.RedirectAfterLogin>
+            {({ action: apiRedirect }) => (
+              <LoginContentWrapper {...props} intl={intl} runtime={runtime} apiRedirect={apiRedirect} />
+            )}
+          </AuthServiceLazy.RedirectAfterLogin>
```
Comment (rnbez) [inline react/components/LoginContent.js:469]: why not the hook version?

File: react/utils/redirect.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export const getDefaultRedirectUrl = isHeaderLogin => {
+  if (!window || !window.location) {
+    return null
+  }
+  if (isHeaderLogin) {
+    return `${window.location.pathname}${window.location.search}`
+  }
+  return '/'
```
Comment (rnbez) [inline react/utils/redirect.js:15]: rootpath?

File: react/components/LoginContent.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rafarubim) [reply react/components/LoginContent.js]: No! It should execute in the browser.

File: react/components/LoginContent.js (hunk 4 above)
Comment (rafarubim) [reply react/components/LoginContent.js]: Next PR!

File: react/components/LoginContent.js (hunk 5 above)
Comment (rafarubim) [reply react/components/LoginContent.js:469]: The hook version was refactored and got a breaking change. I need to change this after that is released

File: react/utils/redirect.js (hunk 6 above)
Comment (rafarubim) [reply react/utils/redirect.js:15]: Next PR :)

File: react/components/LoginContent.js (hunk 3 above)
Comment (rafarubim) [reply react/components/LoginContent.js]: `query` is not used anymore. Does this still apply?

Comment (rnbez) [reply react/components/LoginContent.js]: no

**Pull request merged**